src/frontend/packages/lib_video/src/components/common/Player/videojs/p2pHlsPlugin/index.ts

```diff
@@ -0,0 +1,92 @@
+import videojsHlsjsSourceHandler from '@streamroot/videojs-hlsjs-plugin';
+import { Byterange, Engine } from 'p2p-media-loader-hlsjs';
+import videojs, { VideoJsPlayer } from 'video.js';
+
+import { ExtendedVideoJs, HlsData } from './types';
+
+const Plugin = videojs.getPlugin('plugin');
+
+/**
+ * A VideoJS Plugin enabling the P2P for HLS.
+ *
+ * If different players are downloading the same HLS, some segments may now
+ * be shared to other players through P2P.
+ *
+ * The library doing this is `p2p-media-loader-core`. It uses
+ * WebTorrent trackers servers and a STUN servers (By default it uses the public ones).
+ *
+ * See https://github.com/Novage/p2p-media-loader
+ *
+ * The `videojs-hlsjs-plugin` library is a plugin that will add
+ * necessary metadata to make `p2p-media-loader-hlsjs` works.
+ * @class P2pPlugin
+ */
+export class P2pHlsPlugin extends Plugin {
+  constructor(player: videojs.Player, options: unknown) {
+    const engine = new Engine();
+
+    engine.on('peer_connect', (peer: { id: string; remoteAddress: string }) =>
+      console.log('peer_connect', peer.id, peer.remoteAddress),
```

Comment:
IMO, you should put the log only in dev mode (process.env.NODE_ENV === 'development').

Reply:
This will only work in the standalone site I think, not in LTI.